qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4337,298 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+
+        # XXX actual test here: test that unpurged subvol is purged
+        self.config_set('mgr', self.conf_opt, False)
+        self.wait_for_trashcan_to_be_empty()
+
+    def _get_trashed_sv_path(self, sv, sv_path):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+            strip()
+
+        trashed_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trashed_sv_path = trashed_sv_path.replace(sv, '')
+        return trashed_sv_path
+
+    def _get_num_of_files_in_trashed_sv(self, trashed_sv_path):
+        trashed_sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+        return len(trashed_sv_files)
+
+    def _assert_trashed_sv_has_num_of_files(self, trashed_sv_path,
+                                            num_of_files):
+        sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+
+        self.assertEqual(len(sv_files),  num_of_files)
+
+    def test_when_changed_to_true_ongoing_purge_halts(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        set to true, running "ceph fs subvolume rm" will move the subvolume
+        to trash but (asynchronous) purge of the subvolume won't happen till
+        this config option is not explicitly set to False.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        self.config_set('mgr', self.conf_opt, False)
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        sv_path = self._get_sv_path(v, sv)
+        # adding more files for this test since in once few runs it fails due
+        # to race condition
+        self._do_subvolume_io(sv, number_of_files=200)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        # XXX actual test here: test that purging halts
+        self.config_set('mgr', self.conf_opt, True)
+        trashed_sv_path = self._get_trashed_sv_path(sv, sv_path)
+        NUM_OF_FILES = self._get_num_of_files_in_trashed_sv(trashed_sv_path)
+        time.sleep(2)
+        self._assert_trashed_sv_has_num_of_files(trashed_sv_path,
+                                                 NUM_OF_FILES)
+
+    def test_when_changed_to_false_partly_purged_subvol_purges_fully(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        changed to false, the async purging of a subvolume will resume and
+        also finish, causing trash to be empty.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, number_of_files=100)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self.config_set('mgr', self.conf_opt, True)
+        time.sleep(2)
+
+        # XXX actual test here: test that purging is resumed and finished
+        self.config_set('mgr', self.conf_opt, False)
```

Comment:
Shouldn't we `self._assert_trashed_sv_has_num_of_files(trashed_sv_path, NUM_OF_FILES)` to make sure that it is not empty before purging resumes

Reply:
I don't see the context of this review comment since quoted code is -

```
        time.sleep(2)

        # XXX actual test here: test that purging is resumed and finished
        self.config_set('mgr', self.conf_opt, False)
```

Reply:
Same for this one, the code has moved/removed since this review comment was posted.

Reply:
> I meant to say we should check that trash is not empty before setting the config option to False, then only it make sense to resume purging and deleting non-empty trash dir

The trash would surely empty. so it's not a worry.

Reply:
Marking this as resolved.